vtex-sites/base.store
# chore: Add support for TypeScript files to configure GraphQL servers (`/src/server` folder)

## What's the purpose of this pull request?

This PR aims to increase the usage of TypeScript in the codebase, and the main feature added was the transition from using `.js` files in the `/src/server` folder to using `.ts` files. This should enable users to extend the default server configuration and GraphQL Schema using TypeScript files, which are not supported out of the box today.

This PR also allows us to use TypeScript in the following Gatsby specific files:

- `gatsby-config`
- `gatsby-ssr`
- `gatsby-browser`

Some minor changes also implemented:

- Added a missing developer dependency: `@lhci/cli`. This should be present in the `package.json` file since our `lhci` script depends on it.
- Renames the `src/components/ThirdyPartyScripts` folder to `ThirdPartyScripts`.
- Fixed type inconsistencies between types defined in `@faststore/sdk` and how they were being used here.

## How to test it?

Since this is more of a structural change, to make sure everything is working, all PR checks should be working and the deploy preview should be no different from what's currently live.

## Checklist

<em>You may erase this after checking them all ;)</em>

- [] CHANGELOG entry added
